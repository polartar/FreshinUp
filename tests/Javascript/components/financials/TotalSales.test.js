import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Component from '~/components/financials/TotalSales.vue'
import { FIXTURE_PAYMENT_TYPE_TOTALS } from 'tests/__data__/paymentTypesTotal'

describe('financials/TotalSales', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('required props set', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          gross: 20000,
          net: 10000,
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Visual', () => {
    beforeEach(() => {
      localVue = createLocalVue()
      localVue.use(Vuetify)
    })
    test('gross, net, cash and credit displayed with currency format', () => {
      const component = mount(Component, {
        propsData: {
          gross: 20000,
          net: 10000,
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        },
        localVue
      })
      // gross
      expect(component.text()).toContain('$200.00')
      // net
      expect(component.text()).toContain('$100.00')
      // cash
      expect(component.text()).toContain('$200.00')
      // credit
      expect(component.text()).toContain('$1,050.00')
    })
  })
  describe('Computed', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('cash with set array', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          gross: 20000,
          net: 10000,
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        }
      })
      expect(wrapper.vm.cash).toBe(20000)
    })
    test('cash with empty array', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          gross: 20000,
          net: 10000,
          paymentTypeTotals: []
        }
      })
      expect(wrapper.vm.cash).toBe(0)
    })
    test('credit with set array', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          gross: 20000,
          net: 10000,
          paymentTypeTotals: FIXTURE_PAYMENT_TYPE_TOTALS
        }
      })
      expect(wrapper.vm.credit).toBe(105000)
    })
    test('credit with empty array', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          gross: 20000,
          net: 10000,
          paymentTypeTotals: []
        }
      })
      expect(wrapper.vm.credit).toBe(0)
    })
  })
})
