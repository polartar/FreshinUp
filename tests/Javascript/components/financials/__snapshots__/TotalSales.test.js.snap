// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`financials/TotalSales Snapshots required props set 1`] = `
<div
  class="py-2"
>
  <div
    class="text-xs-center my-5"
  >
    <div
      class="display-1 primary--text"
    >
      
      $200.00
    
    </div>
     
    <div
      class="font-weight-bold subheading mt-2"
    >
      
      Gross
    
    </div>
  </div>
   
  <div
    class="text-xs-center my-5"
  >
    <div
      class="display-1 primary--text"
    >
      
      $100.00
    
    </div>
     
    <div
      class="font-weight-bold subheading mt-2"
    >
      
      Net
    
    </div>
  </div>
   
  <div
    class="text-xs-center my-5"
  >
    <div
      class="display-1 primary--text"
    >
      
      $200.00
    
    </div>
     
    <div
      class="font-weight-bold subheading mt-2"
    >
      
      Cash
    
    </div>
  </div>
   
  <div
    class="text-xs-center my-5"
  >
    <div
      class="display-1 primary--text"
    >
      
      $1,050.00
    
    </div>
     
    <div
      class="font-weight-bold subheading mt-2"
    >
      
      Credit
    
    </div>
  </div>
</div>
`;
